examples/gym/solve_gridworld.py

```diff
@@ -0,0 +1,201 @@
+import logging
+import time
+import sys
+from typing import Tuple
+
+import numpy as np
+from sqlalchemy import create_engine
+
+from rlmolecule.tree_search.reward import Reward
+from rlmolecule.alphazero.tfalphazero_problem import TFAlphaZeroProblem
+from rlmolecule.gym.gym_problem import GymProblem
+from rlmolecule.gym.gym_state import GymEnvState
+from rlmolecule.gym.alphazero_gym import AlphaZeroGymEnv
+
+from examples.gym.tf_model import policy_model_2
+import examples.gym.gridworld_env as gw
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+# NOTE: These class definitions need to stay outside of construct_problem
+# or you will error out on not being able to pickle/serialize them.
+class GridWorldEnv(AlphaZeroGymEnv):
+    def __init__(self,
+                 configured_env: AlphaZeroGymEnv,
+                 **kwargs):
+        super().__init__(configured_env, **kwargs)
+
+    def reset(self):
+        return self.env.reset()
+
+    def get_obs(self) -> np.ndarray:
+        return self.env.get_obs()
+
+
+class GridWorldProblem(GymProblem, TFAlphaZeroProblem):
+    def __init__(self, *,
+                 env: gw.GridEnv,
+                 engine: "sqlalchemy.engine.Engine",
+                 reward_class: Reward,
+                 **kwargs) -> None:
+        super().__init__(reward_class=reward_class, engine=engine, env=env, **kwargs)
```

Comment:
i don't think this `__init__` definition actually doing anything, since it just passes the kwargs to the lower classes? I think this should run if you don't define an `__init__` here.